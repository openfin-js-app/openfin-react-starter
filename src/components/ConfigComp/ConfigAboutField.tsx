import * as React from 'react';
import { WithStyles, withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

import { configAboutFieldStyle as style } from '../../assets/jss/openfin-starter';
import appLogo from '../../assets/svg/app.svg';
import companyLogo from '../../assets/svg/company.svg';

interface IProps extends WithStyles<any>{

}

class ConfigAboutField extends React.Component<IProps,{}>{

    handleReloadBtnClick = ()=>{
        location.reload();
    }

    handleSupportBtnClick = ()=>{
        window.location.href = "mailto:mail@liwentao90@yahoo.com"
    }


    render(){

        const { classes } = this.props;

        return (<React.Fragment>
            <div className={classes.outmostContainer}>
                <div className={classes.imgContainer}>
                    <img className={classes.appImg} src = {appLogo}/>
                    <img className={classes.companyImg} src = {companyLogo}/>
                </div>
                <Typography variant="h3">
                    Openfin react starter
                </Typography>
                <Typography variant="h5">
                    React based openfin desktop template
                </Typography>
                <Typography variant="h6">
                    Version {process.env.REACT_APP_VERSION}
                </Typography>

                <div className={classes.btnContainer}>
                    <Button variant="contained" color='primary'
                            onClick={this.handleReloadBtnClick}
                    >Reload</Button>
                    <Button variant="contained" color='secondary'
                            onClick={this.handleSupportBtnClick}
                    >Support</Button>
                </div>
                <Typography variant={"subtitle1"}>
                    Wentao Li Â© 2018
                </Typography>

            </div>
        </React.Fragment>)

    }
}

export default withStyles(style)(ConfigAboutField);