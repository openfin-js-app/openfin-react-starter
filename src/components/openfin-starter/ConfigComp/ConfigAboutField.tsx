import * as React from 'react';
import { makeStyles } from '@material-ui/styles';

import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

import { useTranslation } from 'react-i18next';

import { configAboutFieldStyle as style } from '../../../assets/jss/openfin-starter';
import appLogo from '../../../assets/svg/app.svg';
import companyLogo from '../../../assets/svg/company.svg';

const useStyles = makeStyles(style);

const ConfigAboutField:React.FunctionComponent<{}> = (
    {}
)=>{

    const classes = useStyles();

    const { t, i18n } = useTranslation('landing', { useSuspense: false });

    const handleReloadBtnClick = ()=>{
        location.reload();
    }

    const handleSupportBtnClick = ()=>{
        window.location.href = "mailto:mail@liwentao90@yahoo.com"
    }

    return (<React.Fragment>
        <div className={classes.outmostContainer}>
            <div className={classes.imgContainer}>
                <img className={classes.appImg} src = {appLogo}/>
                <img className={classes.companyImg} src = {companyLogo}/>
            </div>
            <Typography variant="h3">
                {t('appName')}
            </Typography>
            <Typography variant="h5">
                {t('appDesc')}
            </Typography>
            <Typography variant="h6">
                {t('version')} {process.env.REACT_APP_VERSION}
            </Typography>

            <div className={classes.btnContainer}>
                <Button variant="contained" color='primary'
                        onClick={handleReloadBtnClick}
                >{t('reload')}</Button>
                <Button variant="contained" color='secondary'
                        onClick={handleSupportBtnClick}
                >{t('support')}</Button>
            </div>
            <Typography variant={"subtitle1"}>
                {t('copyright')}
            </Typography>

        </div>
    </React.Fragment>)
}

export default ConfigAboutField;