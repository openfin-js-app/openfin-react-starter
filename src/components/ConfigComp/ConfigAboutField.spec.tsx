import * as React from 'react';
import toJson from 'enzyme-to-json';
import { createShallow, createMount } from '@material-ui/core/test-utils';

import Button from '@material-ui/core/Button';

import ConfigAboutField from './ConfigAboutField';

describe('ConfigAboutField comp', function () {

    let shallow;
    let mount;

    beforeAll(() => {
        mount = createMount();
    });

    afterAll(() => {
        mount.cleanUp();
    });

    beforeEach(() => {
        shallow = createShallow();
    });

    it('renderred correctly',()=>{
        const wrapper = shallow(<ConfigAboutField/>)
        const comp = wrapper.dive();
        expect(toJson(comp)).toMatchSnapshot();
    })

    it('function correctly',()=>{

        let oldLocation = window.location;

        delete window.location;
        window.location = {} as any;

        location.reload = jest.fn();

        window.location.assign = jest.fn((l)=>{
            expect(l).toMatchSnapshot();
        })

        const wrapper = mount(<ConfigAboutField/>);
        expect(wrapper.find(Button)).toHaveLength(2);
        wrapper.find(Button).forEach((button)=>{
            const props = button.props();
            if ( typeof props.onClick === 'function'){
                props.onClick();
            }
        });

        expect(location.reload).toHaveBeenCalled();

        window.location = oldLocation;

    })

});