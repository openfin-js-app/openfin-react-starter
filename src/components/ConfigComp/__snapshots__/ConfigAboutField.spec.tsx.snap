// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigAboutField comp renderred correctly 1`] = `
<Fragment>
  <div
    className="ConfigAboutField-outmostContainer-1"
  >
    <div
      className="ConfigAboutField-imgContainer-2"
    >
      <img
        className="ConfigAboutField-appImg-3"
        src="app.svg"
      />
      <img
        className="ConfigAboutField-companyImg-4"
        src="company.svg"
      />
    </div>
    <WithStyles(Typography)
      variant="h3"
    >
      Openfin react starter
    </WithStyles(Typography)>
    <WithStyles(Typography)
      variant="h5"
    >
      React based openfin desktop template
    </WithStyles(Typography)>
    <WithStyles(Typography)
      variant="h6"
    >
      Version 
      0.30.10
    </WithStyles(Typography)>
    <div
      className="ConfigAboutField-btnContainer-5"
    >
      <WithStyles(Button)
        color="primary"
        onClick={[Function]}
        variant="contained"
      >
        Reload
      </WithStyles(Button)>
      <WithStyles(Button)
        color="secondary"
        onClick={[Function]}
        variant="contained"
      >
        Support
      </WithStyles(Button)>
    </div>
    <WithStyles(Typography)
      variant="subtitle1"
    >
      Wentao Li Â© 2018
    </WithStyles(Typography)>
  </div>
</Fragment>
`;
